backend/tests/unit/core/schema_manager/test_manager_schema.py

```diff
@@ -603,6 +603,69 @@ def extract_weights(schema: SchemaBranch):
     assert len(in_second) == 1 and in_second[0].startswith(new_attr2_partial_id)
 
 
+def test_schema_branch_processes_generic_template_schema_weight(register_core_models_schema):
+    schema = {
+        "generics": [
+            {
+                "name": "GenericDevice",
+                "namespace": "Dcim",
+                "description": "Generic Device object.",
+                "label": "Device",
+                "icon": "mdi:server",
+                "human_friendly_id": ["name__value"],
+                "order_by": ["name__value"],
+                "display_labels": ["name__value"],
+                "generate_template": True,
+                "attributes": [
+                    {"name": "name", "kind": "Text", "unique": True, "order_weight": 7000},
+                    {"name": "description", "kind": "Text", "optional": True, "order_weight": 8000},
+                    {"name": "os_version", "kind": "Text", "optional": True, "order_weight": 5200},
+                ],
+                "relationships": [
+                    {
+                        "name": "tags",
+                        "peer": "BuiltinTag",
+                        "optional": True,
+                        "cardinality": "many",
+                        "kind": "Attribute",
+                        "order_weight": 3000,
+                    },
+                ],
+            },
+            core_object_template,
+            core_object_component_template,
+        ],
+        "nodes": [
+            {
+                "name": "Tag",
+                "namespace": "Builtin",
+                "label": "Tag",
+                "default_filter": "name__value",
+                "attributes": [
+                    {"name": "name", "kind": "Text", "label": "Name", "unique": True},
+                    {
+                        "name": "description",
+                        "kind": "Text",
+                        "label": "Description",
+                        "optional": True,
+                        "branch": BranchSupportType.AGNOSTIC.value,
+                    },
+                ],
+            },
+        ],
+    }
+    schema_branch = SchemaBranch(cache={}, name="test")
+    schema_branch.load_schema(schema=SchemaRoot(**schema))
+    schema_branch.process()
+
+    template = schema_branch.get(name="TemplateDcimGenericDevice", duplicate=False)
+
+    assert template.get_attribute(name="template_name").order_weight == 1000
+    assert template.get_attribute(name="description").order_weight == 2000
+    assert template.get_attribute(name="os_version").order_weight == 3000
```

Comment:
In this case the expectation would be that the order_weight for os_version matches what was set on the generic itself i.e. 5200.